### PR TITLE
Pipeline Launch - reference files

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
image

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [ ] Tests added (or description of how to test) for any new features.
* [ ] User documentation updated for UI or technical changes.

### CODE REVIEW

**Review by deepsidhu85 — COMMENTED** [review]
Still need to review the code but a few things off the bat. The image and text `No reference files can be found within the projects currently in the cart` seem to be taking up a lot of vertical space really for no reason. Maybe it would be better to add an info alert with the text rather than having the image there. Also, I think it would be better to use the `notification` from ant design which pops up from the side rather than the small little notification that pops up at the top center

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java [hunk 1]
```diff
@@ -74,8 +92,21 @@ public UIPipelineDetailsResponse getPipelineDetails(UUID id, Locale locale) thro
          */
 		detailsResponse.setSavedPipelineParameters(getSavedPipelineParameters(workflow, locale));
 
-		return detailsResponse;
-	}
+		/*
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:95]: Fix formatting of this comment

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java [hunk 2]
```diff
@@ -74,8 +92,21 @@ public UIPipelineDetailsResponse getPipelineDetails(UUID id, Locale locale) thro
...
+        Check / add reference files
+         */
+        if (description.requiresReference()) {
+            /*
+            Need to get a list of all the projects in the cart
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:100]: This comment needs to be fixed as well. Looks like there is a missing `*` before the comment text

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java [hunk 3]
```diff
@@ -201,21 +232,42 @@ private String localizedParamOptionLabel(Locale locale, String workflowName, Str
...
+					for (Join<Project, ReferenceFile> projectReferenceFileJoin : referenceFileService.getReferenceFilesForProject(
+							project)) {
+						try {
+							ReferenceFile file = projectReferenceFileJoin.getObject();
+							Path path = file.getFile();
+							String filesize = FileUtilities.humanReadableByteCount(Files.size(path), true);
+							UIReferenceFile uiReferenceFile = new UIReferenceFile(projectReferenceFileJoin, filesize);
+							list.add(uiReferenceFile);
+						} catch (IOException e) {
+							e.printStackTrace();
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:265]: Add a logger to this file and log the stacktrace at the `error` level

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectReferenceFileService.java [hunk 4]
```diff
@@ -51,10 +55,8 @@ public UIProjectReferenceFileService(ProjectService projectService, ReferenceFil
 	 * @throws UnsupportedReferenceFileContentError if content is invalid
 	 * @throws IOException if there is an I/O error
 	 */
-	public String addReferenceFileToProject(Long projectId, List<MultipartFile> files, final Locale locale) throws UnsupportedReferenceFileContentError, IOException {
-		Project project = projectService.read(projectId);
-		logger.debug("Adding reference file to project " + projectId);
-
+	public List<UIReferenceFile> addReferenceFileToProject(Long projectId, List<MultipartFile> files, final Locale locale) throws UnsupportedReferenceFileContentError, IOException {
+		List<UIReferenceFile> uiFiles = new ArrayList<>();
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectReferenceFileService.java:59]: Rename `uiFiles` to `uiReferenceFiles` for clarity

File: src/main/webapp/resources/js/pages/launch/components/UploadReferenceFile.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    onChange(info) {
+      const { status } = info.file;
+      if (status === "done") {
+        message.success(i18n("UploadReferenceFile.success", info.file.name));
+        dispatchReferenceFileUploaded({
+          name: info.file.name,
+          id: info.file.response.id,
+        });
+      } else if (status === "error") {
+        message.error(`${info.file.name} file upload failed.`);
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/launch/components/UploadReferenceFile.jsx:28]: Internationalize

File: src/main/webapp/resources/js/pages/launch/launch-context.js [hunk 6]
```diff
@@ -126,6 +146,7 @@ function LaunchProvider({ children }) {
   }
 
   function dispatchLaunch(values) {
+    console.log(values);
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/launch/launch-context.js:149]: Remove logging statement

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Looking pretty good. A few more comments to add to the previous ones

**Review by deepsidhu85 — COMMENTED** [review]
Also, I don't see any tests for the UIPipelineService. Are you adding them in a later PR?

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java (hunk 1 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:95]: What's wrong with it?

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java (hunk 2 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:100]: I see your issue.  Those are not incorrectly formatted comments, they are just considered "multiline" comments in Java.  They are not supoosed to be JavaDoc.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java (hunk 3 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java]: Added in 3d580f13b80597ef3efffc931916a6a89528f970

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectReferenceFileService.java (hunk 4 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectReferenceFileService.java]: Updated in 30597c3a5fc6ba76574a2514eb8720864089a739

File: src/main/webapp/resources/js/pages/launch/components/UploadReferenceFile.jsx (hunk 5 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/components/UploadReferenceFile.jsx]: Updated in a10eeec132bd8397805a30a186862baf0ab6a576

File: src/main/webapp/resources/js/pages/launch/launch-context.js (hunk 6 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/launch-context.js]: I want to keep this here for now as it is going to be replaced in the final PR for the UI portion of this series.  I have it so I always know what is in that object.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java (hunk 1 above)
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:99]: It was a multi-line comment. Sorry about that

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java (hunk 2 above)
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:104]: Makes sense!

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java [hunk 7]
```diff
@@ -26,15 +39,24 @@
  */
 @Component
 public class UIPipelineService {
-	private final IridaWorkflowsService workflowsService;
-	private final WorkflowNamedParametersService namedParametersService;
+	private static final Logger logger = LoggerFactory.getLogger(UIPipelineService.class);
+
+	private final Cart cart;
+    private final IridaWorkflowsService workflowsService;
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:45]: Formatting

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Looks good. Just one last change and it should be good to go! Great work!

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**